src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.SslCtx.cs

```diff
@@ -165,7 +166,7 @@ internal void EnableSessionCache()
             Interop.Ssl.SslCtxSetData(this, (IntPtr)_gch);
         }
 
-        internal bool TryAddSession(IntPtr namePtr, IntPtr session)
+        internal unsafe bool TryAddSession(IntPtr namePtr, IntPtr session)
```

Comment:
Change the type of namePtr argument to `byte*`, and update the signature of `SslGetServerName` to return `byte*` as well to avoid casts.